src/goose.rs

```diff
@@ -913,6 +910,153 @@ impl GooseUser {
         Ok(single_user)
     }
 
+    /// Returns a reference to GooseUser's session data.
+    /// Leaves the session in-place, creating a new one Option with a reference
+    /// to the original session data.
+    ///
+    /// Return none if no session data has been set or that the session data is not of type `T`
+    ///
+    /// # Example
+    /// ```rust
+    /// use goose::prelude::*;
+    ///
+    /// #[derive(Debug)]
+    /// struct Foo(String);
+    ///
+    /// let mut task = task!(get_session_data_function);
+    ///
+    /// /// A very simple task that makes a GET request.
+    /// async fn get_session_data_function(user: &mut GooseUser) -> GooseTaskResult {
+    ///     let foo = user.get_session_data::<Foo>().expect("Missing session data!");
+    ///     println!("Session data: {}", foo.0);
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_session_data<T: GooseUserData>(&self) -> Option<&T> {
+        match &self.session_data {
+            Some(data) => data.downcast_ref::<T>(),
+            None => None,
+        }
+    }
+
+    /// Returns a reference to GooseUser's session data, without doing bounds
+    /// checking.
+    ///
+    /// For a safe alternative see [`get_session_data`].
+    ///
+    /// # Safety
+    ///
+    /// Calling this method on an GooseUser without session data or with a different type `T` is *[undefined behavior]*
+    ///
+    /// # Example
+    /// ```rust
+    /// use goose::prelude::*;
+    ///
+    /// #[derive(Debug)]
+    /// struct Foo(String);
+    ///
+    /// let mut task = task!(get_session_data_uncheck_function);
+    ///
+    /// /// A very simple task that makes a GET request.
+    /// async fn get_session_data_uncheck_function(user: &mut GooseUser) -> GooseTaskResult {
+    ///     let foo = user.get_session_data_uncheck::<Foo>();
+    ///     println!("Session data: {}", foo.0);
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn get_session_data_uncheck<T: GooseUserData>(&self) -> &T {
```

Comment:
It seems the standard naming for helpers that don't do any checking is `_unchecked()` not `_uncheck()`:
https://doc.rust-lang.org/std/?search=uncheck

For consistency with Rust's std lib please rename this to `get_session_data_unchecked`.